### PR TITLE
Spiral Traversal of Binary Tree

### LINKED ISSUE #853: Binary tree level order traversal in spiral form
Would like to add an algorithm for printing level order traversal of tree in spiral form in C++. 
Please assign this to me under gssoc2021.

### PR DESCRIPTION
**Pull request by tk315**
Kindly check it out and do the needful.
Thanks!

Fixed #853

### CODE REVIEW

File: Code/C++/spiral_traversal_of_binary_tree.cpp [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+After printing we will make the flag reverse and resize the vector to 0. We will keep on doing this process and at the end when the queue becomes empty
+we will get the spiral traversal printed.  
+*/
+
+
+#include <bits/stdc++.h>
+
+using namespace std;
+
+struct Node {
```
Comment (rudrakshi99) [inline Code/C++/spiral_traversal_of_binary_tree.cpp:12]: use `class` instead of `struct`.

File: Code/C++/spiral_traversal_of_binary_tree.cpp [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+        arr.push_back(temp -> data);
+        if (temp -> left != NULL) q.push(temp -> left);
+        if (temp -> right != NULL) q.push(temp -> right);
+    }
+}
+
+int main() {
+    Node * root = new Node(1);
+    root -> left = new Node(2);
```
Comment (rudrakshi99) [inline Code/C++/spiral_traversal_of_binary_tree.cpp:70]: Make it user input by using  `loop` to insert the data.

**Review by rudrakshi99 — CHANGES_REQUESTED** [review]

**Comment by rudrakshi99** [comment]
also use `nullptr` instead of `NULL`

**Comment by tk315** [comment]
All the required changes done.
Kindly review it.

**Review by rudrakshi99 — APPROVED** [review]

**Comment by rudrakshi99** [comment]
@tk315 update the list in the tree folder.

**Comment by tk315** [comment]
@rudrakshi99 Already done

File: Code/C++/spiral_traversal_of_binary_tree.cpp [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+      1         <-
+     / \
+    5   7       ->
+       / \
+      6   4     <-
+         / \
+        2   3   ->
+
+Output:
+1 5 7 4 6 2 3
```
[the remark this answers is not in the document]
Comment (tk315) [reply Code/C++/spiral_traversal_of_binary_tree.cpp]: Requested Changes done

**Comment by rudrakshi99** [comment]
Please resolve merge conflicts.

**Review by OjusWiZard — APPROVED** [review]
Looks good to me

**Pull request merged**